main/src/ca/uwaterloo/flix/api/lsp/provider/InlayHintProvider.scala

```diff
@@ -13,15 +13,87 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package ca.uwaterloo.flix.api.lsp.provider
 
-import ca.uwaterloo.flix.api.lsp.{InlayHint, Range}
+package ca.uwaterloo.flix.api.lsp.provider
+import ca.uwaterloo.flix.api.lsp.acceptors.{FileAcceptor}
+import ca.uwaterloo.flix.language.ast.shared.SymUse
+import ca.uwaterloo.flix.language.ast.TypedAst.{Expr, Root}
+import ca.uwaterloo.flix.language.ast.Type
+import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.api.lsp.{Consumer, InlayHint, InlayHintKind, Position, Range}
+import ca.uwaterloo.flix.api.lsp.Visitor
+import scala.collection.immutable.Map
 
 object InlayHintProvider {
+  private val EnableEffectHints: Boolean = false
 
-  def getInlayHints(uri: String, range: Range): List[InlayHint] = {
-    // We currently do not support any inlay hints.
-    Nil
+  def getInlayHints(uri: String, range: Range)(implicit root: Root): List[InlayHint] = {
+    var inlayHints: List[InlayHint] = Nil
```

Comment:
Is mutation needed here? Looks like we can just do an if-else with Nil in one branch.